server/src/services/application/ClassroomMembershipService.ts

```diff
@@ -0,0 +1,291 @@
+import { ClassroomMembershipRepository } from "services/infrastructure/ClassroomMembershipRepository";
+import { ClassroomRepository } from "services/infrastructure/ClassroomRepository";
+import { ApplicationError } from "utils";
+import {
+  addUserToStreamChannel,
+  removeUserFromStreamChannel,
+} from "services/infrastructure/StreamChannelService";
+
+export class ClassroomMembershipService {
+  private membershipRepository: ClassroomMembershipRepository;
+  private classroomRepository: ClassroomRepository;
+
+  constructor() {
+    this.membershipRepository = new ClassroomMembershipRepository();
+    this.classroomRepository = new ClassroomRepository();
+  }
+
+  /**
+   * Create a request to join a classroom
+   */
+  async createJoinRequest(classroomId: number, userId: number) {
+    // Check if the classroom exists
+    const classroom =
+      await this.classroomRepository.findClassroomById(classroomId);
+    if (!classroom) {
+      throw new ApplicationError("Classroom not found", 404);
```

Comment:
think about errorResponse util function from controllers and this one in services